docs/reference/modules/cluster/misc.asciidoc

```diff
@@ -22,6 +22,47 @@ user with access to the <<cluster-update-settings,cluster-update-settings>>
 API can make the cluster read-write again.
 
 
+[[cluster-shard-limit]]
+
+==== Cluster Shard Limit
+
+In a Elasticsearch 7.0 and later, there will be a soft limit on the number of
+shards in a cluster, based on the number of nodes in the cluster.  This is
+intended to prevent operations which may unintentionally destabilize the
+cluster. Until 7.0, actions which would result in the cluster going over the
+limit will issue a deprecation warning.
+
+NOTE: You can set the system property `es.enforce.shard_limit` to `true` to opt
```

Comment:
I don't think we need to namespace this under `enforce`, so `es.enforce.shard_limit` -> `es.enforce_shard_limit`. And perhaps it should more closely reflect the name of the setting, so `es.enforce.shard_limit` -> `es.enforce_max_shards_per_node`.